formula/src/main/java/com/instacart/formula/FormulaRuntime.kt

```diff
@@ -174,26 +174,24 @@ class FormulaRuntime<Input : Any, Output : Any>(
 
     override fun executeBatch(updates: List<() -> Unit>) {
         // Using default dispatcher for batch execution
-        defaultDispatcher.dispatch {
```

Comment:
Instead of dispatching and then posting update, we add update to the queue and dispatch request to process. This way the queue keeps track of all events in arrival order while processing can be shifted to a different thread.